File: back/src/main/java/com/back/domain/review/dto/ReviewDto.java
package com.back.domain.review.dto;

import com.back.domain.member.dto.MemberDto;
import com.back.domain.review.entity.Review;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;
import java.util.List;

public record ReviewDto(
        @NotNull
        Long id,
        @NotNull
        Long bookId,
        @NotNull
        Float rating,
        @NotNull
        String content,
        @NotNull
        LocalDateTime modifiedDate,
        @NotNull
        LocalDateTime createdDate,
        @NotNull
        MemberDto reviewer,
        @NotNull
        List<String> tags
) {
    public ReviewDto(Review review) {
        this(
                review.getId(),
                review.getBook().getId(),
                review.getRating(),
                review.getContent(),
                review.getModifiedDate(),
                review.getCreatedDate(),
                new MemberDto(review.getReviewer()),
                review.getTags());
    }
}

File: back/src/main/kotlin/com/back/domain/review/dto/ReviewDto.kt
package com.back.domain.review.dto

import com.back.domain.member.dto.MemberDto
import com.back.domain.review.entity.Review
import java.time.LocalDateTime

open class ReviewDto(
    val id: Long,
    val bookId: Long,
    val bookTitle: String,
    val rating: Float,
    val content: String,
    val modifiedDate: LocalDateTime,
    val createdDate: LocalDateTime,
    val reviewer: MemberDto,
    val tags: List<String>,
    val likeCount: Int
) {
    constructor(review: Review) : this(
        review.id,
        review.book.id,
        review.book.title,
        review.rating,
        review.content,
        review.modifiedDate,
        review.createdDate,
        MemberDto(review.reviewer),
        review.tags,
        review.likeCount
    )
}
